azure/src/main/java/org/apache/iceberg/azure/adlsv2/ADLSLocation.java

```diff
@@ -53,19 +63,17 @@ class ADLSLocation {
 
     ValidationException.check(matcher.matches(), "Invalid ADLS URI: %s", location);
 
-    String authority = matcher.group(1);
-    String[] parts = authority.split("@", -1);
-    if (parts.length > 1) {
-      this.container = parts[0];
-      this.storageAccount = parts[1];
-    } else {
-      this.container = null;
-      this.storageAccount = authority;
+    try {
+      URI uri = new URI(location);
+      this.container = uri.getUserInfo();
+      // storage account name is the first part of the host
+      int accountSplit = uri.getHost().indexOf('.');
+      String storageAccountName = uri.getHost().substring(0, accountSplit);
+      this.storageAccount = String.format("%s.dfs.core.windows.net", storageAccountName);
```

Comment:
For `wasb`, the original `host` URL is a `blob.core.windows.net` endpoint (see the sample above). Could you clarify if the change from blob to dfs is necessary?

Reply:
I don't believe this is required b/c it appears the `blob` host would be changed in the ADLS client when setting the endpoint [here](https://github.com/Azure/azure-sdk-for-java/blob/fd36287fb689545c91b8417c7fab11d1a9ff64a3/sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClientBuilder.java#L187C38-L187C63).   I'll make an update for this b/c I think we may also want to change some of the nomenclature as well